src/api/goods.ts

```
@@ -1,16 +1,26 @@
 import { Good } from '../types/Good';
 
-// eslint-disable-next-line
-const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
+const API_URL =
+  'https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json';
 
+// завантажує всі товари
 export function getAll(): Promise<Good[]> {
   return fetch(API_URL).then(response => response.json());
 }
 
-export const get5First = () => {
-  return getAll().then(goods => goods); // sort and get the first 5
+// завантажує перші 5 (відсортовані за назвою)
+export const get5First = async (): Promise<Good[]> => {
```

`get5First` and `getRedGoods` currently call `getAll` and do not add their own error handling. According to the implementation requirements (see Implementation Details 3.2: "Handle fetch/network failures. Handle non-2xx HTTP statuses. Return rejected promises or throw errors to be handled by the UI layer"), ensure these functions either rely on `getAll` to throw (acceptable) or explicitly propagate/augment errors instead of swallowing them. If you keep logic as-is, make sure `getAll` throws descriptive errors so these methods will propagate them to the UI.